compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleKind.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.projects;
+
+/**
+ * Represents the kind of the {@code Module} instance.
+ * <p>
+ * There are two known kinds of module at the moment: {user-provided, compiler-generated},
+ *
+ * @since 2.3.0
+ */
+enum ModuleKind {
+    COMPILER_GENERATED,
+    USER_PROVIDED,
+    ;
```

Comment:
Is the semicolon here required?

Reply:
Yes, it denotes the end of the enum values